crates/bifrost/src/providers/replicated_loglet/sequencer/appender.rs

```diff
@@ -353,24 +337,34 @@ impl<'a, T: TransportConnect> LogServerStoreTask<'a, T> {
         let server_local_tail = self
             .server
             .local_tail()
-            .wait_for_offset_or_seal(self.first_offset)
-            .await?;
+            .wait_for_offset_or_seal(self.first_offset);
 
-        match server_local_tail {
+        let global_tail = self
+            .sequencer_shared_state
+            .committed_tail
+            .wait_for_offset_or_seal(self.first_offset);
+
+        let tail_state = tokio::select! {
+            local_state = server_local_tail => {
+                local_state?
+            }
+            global_state = global_tail => {
+                global_state?
+            }
+        };
+
+        match tail_state {
             TailState::Sealed(offset) => return Ok(StoreTaskStatus::Sealed(offset)),
             TailState::Open(offset) => {
                 match offset.cmp(&self.first_offset) {
-                    Ordering::Equal => {
+                    Ordering::Equal | Ordering::Greater => {
```

Comment:
Should we return on line 385 a `StoreTaskStatus::Sealed` if the response status is `Sealing` or `Sealed`?

Reply:
You are correct, thank you for catching this.